pywinauto/macos/keyboard_helper.py

```diff
@@ -0,0 +1,664 @@
+            # hold down the modifier key
+            code_keys,flag_from_func=handle_code(c)
+            if flag_from_func is not None:
+                flag=flag_from_func
+            # keys.append(KeyAction(modifier, up = False))
+            if DEBUG:
+                print("MODS+", modifiers)
+            continue
+
+        # Apply modifiers over a bunch of characters (not just one!)
+        elif c == "(":
+            # find the end of the bracketed text
+            end_pos = string.find(")", index)
+            if end_pos == -1:
+                raise KeySequenceError('`)` not found')
+            keys.extend(parse_keys(string[index:end_pos], modifiers = modifiers,flag=flag))
+            index = end_pos + 1
+
+        # Escape or named key
+        elif c == "{":
+            # We start searching from index + 1 to account for the case {}}
+            end_pos = string.find("}", index+1)
+            if end_pos == -1:
+                raise KeySequenceError('`}` not found')
+
+            code_name = string[index:end_pos]
+            index = end_pos + 1
+            code_keys,flag_from_func=handle_code(code_name)
+            if flag is not None:
+                keys.extend(decode_flag_to_key(flag))
+            flag=flag_from_func
+            if index==len(string) and flag is not None:
+                keys.extend(decode_flag_to_key(flag))
+            keys.extend(code_keys)
+        # unmatched ")"
+        elif c == ')':
+            raise KeySequenceError('`)` should be preceeded by `(`')
+
+        # unmatched "}"
+        elif c == '}':
+            raise KeySequenceError('`}` should be preceeded by `{`')
+        # so it is a normal character
+        else:
+            # don't output white space unless flag to output have been set
+            if (c == ' ' and not with_spaces or \
+                    c == '\t' and not with_tabs or \
+                    c == '\n' and not with_newlines):
+                continue
+
+            # output newline
+            if c in ('~', '\n'):
+                keys.append(KeyAction(selected_keyboard.CODES["ENTER"],flag=flag))
+                flag=None
+            else:
+                if c.isalpha() and c.isupper():
+                    flag=kCGEventFlagMaskShift
+                    c=c.lower()
+                if c in selected_keyboard.shift_chars:
+                    key_code=selected_keyboard.shift_chars[c]
+                    if(key_code in selected_keyboard.key_code_map):
+                        flag=kCGEventFlagMaskShift
+                        key_code=selected_keyboard.key_code_map[key_code]
+                        keys.append(KeyAction(key_code,flag=flag))
+                    flag=None
+                elif(c in selected_keyboard.key_code_map):
+                    key_code=selected_keyboard.key_code_map[c]
+                    keys.append(KeyAction(key_code,flag=flag))
+                    flag=None
+
+    return keys
+
+
+def SendKeys(keys,
```

Comment:
It's worth using `send_keys` name only.

Reply:
@vasily-v-ryabov should I also rename this method in linux/base_wrapper ?

Reply:
Yes, it makes sense.